### ISSUE TITLE
" 'path' must be None or a list, not <class '_frozen_importlib_external._NamespacePath'>"

### ISSUE THREAD

**Issue by chaitusyn** [spyder-ide/spyder#6568]
## Description
I am just importing below and I see this error.
import matplotlib.pyplot as plt
import matplotlib.cm
from mpl_toolkits.basemap import Basemap
from matplotlib.patches import Polygon
from matplotlib.collections import PatchCollection
from matplotlib.colors import Normalize

**What steps will reproduce the problem?**

1. Just start a new script by importing matplotlib
2. 
3. 

**What is the expected output? What do you see instead?**
There should not be any errors. But we got below erorrs
*********************************************************** 
Python 3.6.4 |Anaconda, Inc.| (default, Jan 16 2018, 10:22:32) [MSC v.1900 64 bit (AMD64)]
Type "copyright", "credits" or "license" for more information.

IPython 6.2.1 -- An enhanced Interactive Python.

runfile('C:/Users/g704252/.spyder-py3/untitled.py', wdir='C:/Users/g704252/.spyder-py3')
Traceback (most recent call last):

  File "<ipython-input-1-dc09800cf7ff>", line 1, in <module>
    runfile('C:/Users/g704252/.spyder-py3/untitled.py', wdir='C:/Users/g704252/.spyder-py3')

  File "C:\Users\g704252\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 705, in runfile
    execfile(filename, namespace)

  File "C:\Users\g704252\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 102, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "C:/Users/g704252/.spyder-py3/untitled.py", line 3, in <module>
    from mpl_toolkits.basemap import Basemap

  File "<frozen importlib._bootstrap>", line 971, in _find_and_load

  File "<frozen importlib._bootstrap>", line 951, in _find_and_load_unlocked

  File "<frozen importlib._bootstrap>", line 890, in _find_spec

  File "<frozen importlib._bootstrap>", line 864, in _find_spec_legacy

  File "C:\Users\g704252\AppData\Local\Continuum\anaconda3\lib\site-packages\pyximport\pyximport.py", line 253, in find_module
    fp, pathname, (ext,mode,ty) = imp.find_module(fullname,package_path)

  File "C:\Users\g704252\AppData\Local\Continuum\anaconda3\lib\imp.py", line 271, in find_module
    "not {}".format(type(path)))

RuntimeError: 'path' must be None or a list, not <class '_frozen_importlib_external._NamespacePath'>
*******************************************************************************

**Please provide any additional information below**

## Version and main components

* Spyder Version: 3.2.6 
* Python Version: 3.6.4
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.22.0 (OK)
numpy >=1.7      :  1.14.0 (OK)
sphinx >=0.6.6   :  1.6.6 (OK)
rope >=0.9.4     :  0.10.7 (OK)
jedi >=0.9.0     :  0.11.1 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.27.3 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.2.1 (OK)
pylint >=0.25    :  1.8.2 (OK)

```

**Comment by chaitusyn** [comment]
By seeing other posts I tried changing the dependencies 
Help--> Dependencies--> Copy to Clipboard 
but its not opening anything. SO how do we change?

**Comment by ccordoba12** [comment]
Thanks for reporting. This is a duplicate of issue #5299. Please see there for a temporary fix.

**Comment by chaitusyn** [comment]
Hi , I commented this section of the code in sitecustomize.py 
Still its not working.Could you please help me proceed with this.I am new to Python and still learning stuff
#if HAS_PYXIMPORT:
#    # Add Numpy include dir to pyximport/distutils
#    try:
#        import numpy
#        pyx_setup_args['include_dirs'] = numpy.get_include()
#    except:
#        pass

    # Setup pyximport and enable Cython files reload
#    pyximport.install(setup_args=pyx_setup_args, reload_support=True)
#    
#try:
#    # Import cython_inline for runfile function
#    from Cython.Build.Inline import cython_inline
#    HAS_CYTHON = True
#except:
#    HAS_CYTHON = False

**Comment by chaitusyn** [comment]
Hi , I commented this section of the code in sitecustomize.py
 Still its not working.Could you please help me proceed with this.I am new to Python and still learning stuff
 #if HAS_PYXIMPORT:




#if HAS_PYXIMPORT:
#    # Add Numpy include dir to pyximport/distutils
#    try:
#        import numpy
#        pyx_setup_args['include_dirs'] = numpy.get_include()
#    except:
#        pass

    # Setup pyximport and enable Cython files reload
#    pyximport.install(setup_args=pyx_setup_args, reload_support=True)
#    
#try:
#    # Import cython_inline for runfile function
#    from Cython.Build.Inline import cython_inline
#    HAS_CYTHON = True
#except:
#    HAS_CYTHON = False

**Comment by chaitusyn** [comment]
Hi , I commented this section of the code in sitecustomize.py
 Still its not working.Could you please help me proceed with this.I am new to Python and still learning stuff
 #if HAS_PYXIMPORT:




#if HAS_PYXIMPORT:
#    # Add Numpy include dir to pyximport/distutils
#    try:
#        import numpy
#        pyx_setup_args['include_dirs'] = numpy.get_include()
#    except:
#        pass

    # Setup pyximport and enable Cython files reload
#    pyximport.install(setup_args=pyx_setup_args, reload_support=True)
#    
#try:
#    # Import cython_inline for runfile function
#    from Cython.Build.Inline import cython_inline
#    HAS_CYTHON = True
#except:
#    HAS_CYTHON = False

**Comment by CAM-Gerlach** [comment]
Are you getting the exact same error as before? Did you edit the correct file? At what location? Did you quit and relaunch Spyder? Also, it doesn't look like the try/except block loading ``cython`` has anything to do with ``pyximport``, so you can probably uncomment that...

**Comment by chaitusyn** [comment]
Hi I edited below file : 
C:\Users\xxx\AppData\Local\Continuum\anaconda3\Lib\site-packages\spyder\utils\site\sitecustomize.py

I uncommented CYTHON part as you suggested

```
#if HAS_PYXIMPORT:
#    # Add Numpy include dir to pyximport/distutils
#    try:
#        import numpy
#        pyx_setup_args['include_dirs'] = numpy.get_include()
#    except:
#        pass

    # Setup pyximport and enable Cython files reload
#    pyximport.install(setup_args=pyx_setup_args, reload_support=True)
#    
try:
    # Import cython_inline for runfile function
    from Cython.Build.Inline import cython_inline
    HAS_CYTHON = True
except:
    HAS_CYTHON = False

```

 
Now I am getting below error
****************************************************************
```
File "<ipython-input-3-dc09800cf7ff>", line 1, in <module>
    runfile('C:/Users/g704252/.spyder-py3/untitled.py', wdir='C:/Users/g704252/.spyder-py3')

  File "C:\Users\g704252\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 698, in runfile
    if HAS_CYTHON and os.path.splitext(filename)[1].lower() == '.pyx':

NameError: name 'HAS_CYTHON' is not defined
```
****************************************************************

**Comment by chaitusyn** [comment]
Hi I edited below file : 
C:\Users\xxx\AppData\Local\Continuum\anaconda3\Lib\site-packages\spyder\utils\site\sitecustomize.py

I uncommented CYTHON part as you suggested
```

#if HAS_PYXIMPORT:
#    # Add Numpy include dir to pyximport/distutils
#    try:
#        import numpy
#        pyx_setup_args['include_dirs'] = numpy.get_include()
#    except:
#        pass

    # Setup pyximport and enable Cython files reload
#    pyximport.install(setup_args=pyx_setup_args, reload_support=True)
#    
try:
    # Import cython_inline for runfile function
    from Cython.Build.Inline import cython_inline
    HAS_CYTHON = True
except:
    HAS_CYTHON = False


```
 
Now I am getting below error
****************************************************************
```
File "<ipython-input-3-dc09800cf7ff>", line 1, in <module>
    runfile('C:/Users/xxx/.spyder-py3/untitled.py', wdir='C:/Users/xxx/.spyder-py3')

  File "C:\Users\xxx\AppData\Local\Continuum\anaconda3\lib\site-packages\spyder\utils\site\sitecustomize.py", line 698, in runfile
    if HAS_CYTHON and os.path.splitext(filename)[1].lower() == '.pyx':

NameError: name 'HAS_CYTHON' is not defined

```**************************************************************** 
Sorry about formatting in earlier post. Only after posting i got to see the difference 
I will  take care from next time.

**Comment by CAM-Gerlach** [comment]
Its quite possible I was wrong. In any case, did you get the exact same error as originally when you did comment it out, or what? @ccordoba12 , is the user commenting out the correct code block?
